src/principal/service.ts

```diff
@@ -27,6 +27,22 @@ export async function findAll(): Promise<Principal[]> {
 
 }
 
+export async function getPrincipleStats(): Promise<PrincipalStats> {
+
+  const query = 'SELECT type, COUNT(*) as total FROM principals GROUP BY type';
+  const result = await database.query(query);
+
+  const principalStats: any = {};
+  // const principalStats: Record<PrincipalType, number> = {};
```

Comment:
Is there way to type this object?

Reply:
Type it as `PrincipalStats` and instead of starting with an empty object, start with an object with a 3 properties and a value of `0` for each.